exercises/robot-name/uRobotNameTest.pas

```diff
@@ -0,0 +1,79 @@
+unit uRobotNameTest;
+
+interface
+uses
+  DUnitX.TestFramework, uRobotName;
+
+type
+  [TestFixture]
+  TRobotNameTest = class(TObject)
+  private
+    FRobot : TRobot;
+  public
+    [Test]
+//    [Ignore('Comment the "[Ignore]" statement to run the test')]
+    procedure name_properly_formated;
+
+    [Test]
+    [Ignore]
+    procedure is_name_persistent;
+
+    [Test]
+    [Ignore]
+    procedure is_able_to_reset_name;
+
+    [Test]
+    [Ignore]
+    procedure different_robots_have_different_names;
+  end;
+
+implementation
+
+uses
+  System.RegularExpressions, System.Classes, Generics.Collections;
+
+procedure TRobotNameTest.is_name_persistent;
+begin
+  FRobot := TRobot.Create;
+  Assert.AreEqual(FRobot.Name, FRobot.Name);
+  FRobot.DisposeOf;
+end;
+
+procedure TRobotNameTest.name_properly_formated;
+begin
+  FRobot := TRobot.Create;
+  Assert.IsTrue(TRegEx.IsMatch(FRobot.Name, '^[A-Z]{2}\d{3}$'), 'expected format''ssddd'', found ' + FRobot.Name);
+  FRobot.DisposeOf;
+end;
+
+procedure TRobotNameTest.different_robots_have_different_names;
+var Names : TList<string>;
+  Robots : TObjectList<TRobot>;
+  i: Integer;
+begin
+  Names := TList<string>.create;
+  Robots := TObjectList<TRobot>.Create;
```

Comment:
I think you need to invoke that `Robots` will take ownership of the objects that are being added.  That way the objects are properly destroyed when `Robots` is disposed of.  The way `TObjectList<>` is being used is no different than a `TList<>`

Reply:
Never mind me!  I just looked up TObjectList and am reminded that by DEFAULT it assumes ownership of the objects added to it.  :+1:   Well played sir.